internal/k8s/reconciler/validator/route.go

```diff
@@ -0,0 +1,154 @@
+package validator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/gatewayclient"
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/reconciler/state"
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/service"
+)
+
+// all kubernetes routes implement the following two interfaces
+type Route interface {
+	client.Object
+	schema.ObjectKind
+}
+
+// RouteValidator is responsible for taking a provided v1alpha2.HTTPRoute or
+// v1alpha2.TCPRoute and deriving a state.RouteState from it. Ultimately, this
+// RouteState is what makes up the Status on the Kubernetes Route resource.
```

Comment:
```suggestion
// RouteState is what makes up the Status on the Kubernetes Route resource
// and holds any intermediate state resolved from remote object references
// (i.e. resolved Consul services from `backendRefs` values).
```

Reply:
Also, looks like we have a mismatch for `v1beta1.Gateway` in the above docstring v. `v1alpha2.HTTP/TCPRoute` here.